src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -291,23 +291,39 @@ export function collapseDeepestExpandedLevel(accessor: ServicesAccessor) {
 		 */
 		const navigator = viewer.navigate();
 		let node = navigator.first();
-		let collapseFileMatchLevel = false;
-		if (node instanceof FolderMatch) {
-			while (node = navigator.next()) {
-				if (node instanceof Match) {
-					collapseFileMatchLevel = true;
-					break;
+		let canCollapseFileMatchLevel = false;
+		let canCollapseFirstLevel = false;
+
+		while (node = navigator.next()) {
+			if (node instanceof Match) {
+				canCollapseFileMatchLevel = true;
+				break;
+			}
+			if (searchView.isTreeViewVisible && !canCollapseFirstLevel) {
+				const immediateParent = node.parent();
+				if (!(immediateParent instanceof SearchResult) && !(immediateParent.parent() instanceof SearchResult)) {
```

Comment:
This "two steps away from SearchResult" is a little odd. The workspace folder still won't show up when you only have a single folder open in the window, right?

Reply:
The workspace folder won't appear, but it will still be in the model (in single root). The reason why I did it this way is because it seemed like collapsing at the level right below the root was ideal regardless of whether it was single or multi-root, so it made the most sense to check the parent of the parent for the searchResult. Is there a cleaner way to do this?